include/deal.II/lac/trilinos_vector.h

```diff
@@ -2009,8 +2009,9 @@ namespace TrilinosWrappers
       // if we have ghost values, do not allow
       // writing to this vector at all.
       Assert(!has_ghost_elements(), ExcGhostsPresent());
-      Assert(local_size() == v.local_size(),
-             ExcDimensionMismatch(local_size(), v.local_size()));
+      Assert(locally_owned_size() == v.locally_owned_size(),
+             ExcDimensionMismatch(locally_owned_size(),
+                                  v.locally_owned_size()));
```

Comment:
I know this is pre-existing, but shouldn't we use the equivalent
```suggestion
      AssertDimension(locally_owned_size(), v.locally_owned_size());
```
(here and below)?

Reply:
OK. I have replace all occurrence of `ExcDimensionMismatch`!